drivers/sensor/iis2iclx/iis2iclx_trigger.c

```diff
@@ -238,7 +239,7 @@ int iis2iclx_init_interrupt(const struct device *dev)
 	}
 
 	/* enable interrupt on int1/int2 in pulse mode */
-	if (iis2iclx_int_notification_set(&iis2iclx->ctx,
+	if (iis2iclx_int_notification_set((stmdev_ctx_t *)&cfg->ctx,
```

Comment:
Why do you need the cast here?  `&cfg->ctx` is the right type already

Reply:
It complains with a warning like this

 In file included from /local/home/visconti/Projects/zephyrproject/zephyr/drivers/sensor/iis2iclx/iis2iclx.h:19,
                 from /local/home/visconti/Projects/zephyrproject/zephyr/drivers/sensor/iis2iclx/iis2iclx_trigger.c:18:
/local/home/visconti/Projects/zephyrproject/modules/hal/st/sensor/stmemsc/iis2iclx_STdC/driver/iis2iclx_reg.h:2605:53: note: expected 'stmdev_ctx_t *' {aka 'struct <anonymous> *'} but argument is of type 'const stmdev_ctx_t *' {aka 'const struct <anonymous> *'}

I guess this is because cfg is declared as

 ```const struct iis2iclx_config *cfg```

Isn't it? Is there a different way to avoid the warning?

Reply:
You could fix the prototype to make the argument pointer-to-const.  See also STMicroelectronics/STMems_Standard_C_drivers#121.

Casting away const is a MISRA violation, and generally a bad idea indicating a flawed API.

Reply:
I don't want to change the prototype, because I want to keep the stmemsc module (the STdC driver in our terminology) same as it was released, without any change. So, I asked my colleague  who is actually maintaining the STdC drivers to change it, but I'm not sure he want to do that even if it is correct what you said. Moreover I noticed that you already commented few days ago for the exactly same reason (or maybe similar, I don't recall) in that github repo.

Reply:
OK, I hadn't noticed that function was from stmemc.  In that case I guess you have to have the cast.

Reply:
@pabigot 
https://github.com/STMicroelectronics/STMems_Standard_C_drivers/issues/121

Reply:
Oh, sorry. It is the one that you just mentioned above... :D